tests/system/conftest.py

```diff
@@ -34,3 +35,54 @@ def identity():
         channel=grpc.insecure_channel('localhost:7469'),
     )
     return IdentityClient(transport=transport)
+
+
+class MetadataClientInterceptor(grpc.UnaryUnaryClientInterceptor,
+                                grpc.UnaryStreamClientInterceptor,
+                                grpc.StreamUnaryClientInterceptor,
+                                grpc.StreamStreamClientInterceptor):
+
+    def __init__(self, key, value):
+        self._key = key
+        self._value = value
+
+    def _add_metadata(self, client_call_details):
+        if client_call_details.metadata is not None:
+            client_call_details.metadata.append((self._key, self._value,))
+
+    def intercept_unary_unary(self, continuation, client_call_details, request):
+        self._add_metadata(client_call_details)
```

Comment:
Unless I'm missing something, the code for all the interceptor calls is now identical. Maybe abstract that into a helper method and just alias that.

Reply:
there are slice differences in the ```continuation``` call arg and return value, request vs request_iterator, response/response_iterator. I think it might be more clear to write them separately.